CMakeLists.txt

```diff
@@ -68,15 +68,38 @@ endif()
 
 set(CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake ${CMAKE_MODULE_PATH})
 
-# Enable the built-in updater by default unless NO_INCLUDE_UPDATER is
-# already defined. Linux packagers will find it useful to do this since
-# package managers are responsible for updates there. Automatically exclude
-# the update stuff from FreeBSD and Cygwin because if they ever get finished
-# they have their own packaging system
-if((DEFINED ENV{NO_INCLUDE_UPDATER} AND NOT $ENV{MUDLET_VERSION_BUILD} STREQUAL "") OR (CMAKE_SYSTEM_NAME STREQUAL "FreeBSD") OR (CMAKE_SYSTEM_NAME STREQUAL "Cygwin"))
-  option(WITH_UPDATER "Include libraries and code to allow release versions to be updated on demand" OFF)
+# Enable the built-in updater by default on the three main supported OS
+# unless environmental variable WITH_UPDATER is already defined and set to "NO"
+# (case insensitive).
+# Linux packagers will find it useful to do this since package managers are
+# responsible for updates there. Automatically exclude the update stuff from
+# other OSs such as FreeBSD and Cygwin because if they ever get finished they
+# have their own packaging system - and we do not support updating them.
+# Note: WITH_UPDATER is an environmental value/variable (could be a number, a
+#           string, something else or not even exist).
+#       UPDATER_TEST and USE_UPDATER are cmake variables (possibly a string and
+#           NO(false) or YES(true) respectively).
+#       INCLUDE_UPDATER is a C preprocessor symbol.
```

Comment:
:+1: thanks for revising this! Looks good.

Reply:
Yeah, it took me a while to get them clearly differentiated in my head so this might help others to spot the differences.